pandas/tests/indexes/test_multi.py

```diff
@@ -3291,3 +3291,29 @@ def test_duplicate_multiindex_labels(self):
         with pytest.raises(ValueError):
             ind.set_levels([['A', 'B', 'A', 'A', 'B'], [2, 1, 3, -2, 5]],
                            inplace=True)
+
+    @pytest.mark.parametrize("midx,idx,count", [
+        (pd.MultiIndex.from_product([[0, 1], [1, 0]]), pd.Series(range(4)), 4),
+        (pd.MultiIndex.from_product([[0, 1]]), pd.Series(range(2)), 2)])
+    def test_multiindex_compare(self, midx, idx, count):
+        # GH 21149
+        '''Ensure comparison operations for MultiIndex with nlevels == 1
+            behave consistently with those for MultiIndex with nlevels > 1
+        '''
+        expected = pd.Series([True]).repeat(count)
+        expected.reset_index(drop=True, inplace=True)
+        # Equality self-test: MultiIndex object vs self
+        result = pd.Series(midx == midx)
+        tm.assert_series_equal(result, expected)
+        # Equality self-test: non-MultiIndex Index object vs self
+        result = (idx == idx)
```

Comment:
where did the idea for these test come from? what exactly are you trying to test here?

Reply:
These tests are trying to ensure the behaviour on comparison between `MultiIndex vs MultiIndex`, `MultiIndex vs Index` is consistent, irrespective of `nlevels`. Also, ensuring that the comparison behaviour for `Index vs Index` has not changed due to change in this PR.

Currently (as of 0.23.0), comparing MultiIndex of nlevel==1 with another of same length raises a ValueError e.g.
```python
[In] midx=pd.MultiIndex.from_product([[0, 1]])
[In] midx
[Out] MultiIndex(levels=[[0, 1]],
           labels=[[0, 1]])
[In] midx == midx
[Out] ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

whereas the behaviour should be consistent with that for MultiIndex with nlevel>1 as follows:
```python
[In] midx == midx
[Out] array([ True,  True])
```

Reply:
These tests are trying to ensure the behaviour on comparison between `MultiIndex vs MultiIndex`, `MultiIndex vs Index` is consistent, irrespective of `nlevels`. Also, ensuring that the comparison behaviour for `Index vs Index` has not changed due to change in this PR.

Currently (as of 0.23.0), comparing MultiIndex of nlevel==1 with another of same length raises a ValueError e.g.
```python
[In] midx=pd.MultiIndex.from_product([[0, 1]])
[In] midx
[Out] MultiIndex(levels=[[0, 1]],
           labels=[[0, 1]])
[In] midx == midx
[Out] ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```
whereas the behaviour should be consistent with that for MultiIndex with nlevel>1 as follows:
```python
[In] midx == midx
[Out] array([ True,  True])
```

Reply:
in this test, probably should be OK to only keep the `midx == midx` and `midx > midx` checks